File: 49-GroupAnagrams/49-GroupAnagrams.py
# Last updated: 6/25/2026, 9:17:29 AM
from collections import defaultdict
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        d= defaultdict(list)

        for s in strs:
            arr=[0]*26
            for c in s:
                arr[ord(c)-ord('a')]+=1
            d[tuple(arr)].append(s)

            #d[''.join(sorted(s))].append(s)
        
        return list(d.values())

File: 49-GroupAnagrams/49-GroupAnagrams.cpp
// Last updated: 6/25/2026, 9:17:33 AM
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> result;
        unordered_map<string, vector<string>> umap;
        if(strs.size()==0) return result;
        for(int i=0; i<strs.size(); i++){
            string s = strs[i];
            sort(s.begin(), s.end());
            umap[s].push_back(strs[i]);
        }
        for(auto i: umap){
            result.push_back(i.second);
        }
        return result;
    }
};